src/LabLog/Controllers/AdminController.cs

```diff
@@ -12,31 +12,32 @@ namespace LabLog.Controllers
 {
     public class AdminController : Controller
     {
-        public IActionResult Index(string id)
 
+        public IActionResult Index()
         {
-            using (var db = new RoomModelContext())
-            {
-                    ViewData["Rooms"] = db.Rooms.ToList();
-            }
+            ViewData["Rooms"]="";
             return View();
         }
 
+
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public IActionResult Index(RoomModel room)
+        public IActionResult Index(Room formroom)
         {
-            if (ModelState.IsValid)
-            {
-                Console.WriteLine(room.Name + " added to database");
-                using (var db = new RoomModelContext())
+            //if (ModelState.IsValid)
+            //{
+                // Use this so attempt to add to database only occurs when model is valid.
+            //}
+            int count;
+            var room = LabLog.Domain.Entities.Room.Create(e => {
+                using (var db = new EventModelContext())

```

Comment:
definitely do not new up the eventmodelcontext here
it should be injected into the controller's constructor

Reply:
I don't know how to do that. Where is the controller called from? How do I inject it? I can create the constructor for the controller, but don't know how to send the eventmodelcontext to it.